packages/template-retail-react-app/app/hooks/useAutocompleteSuggestions.js

```
@@ -0,0 +1,134 @@
+
+    // indicates if there is currently an incomplete request to the places API
+    const [isLoading, setIsLoading] = useState(false)
+
+    // Debounce timeout ref
+    const debounceTimeoutRef = useRef(null)
+
+    // Fetch suggestions from Google Maps API
+    const fetchSuggestions = useCallback(
+        async (input) => {
+            if (!places || !apiKey || !input || input.length < 3) {
+                setSuggestions([])
+                return
+            }
+
+            setIsLoading(true)
+
+            try {
+                const {AutocompleteSessionToken, AutocompleteSuggestion} = places
+
+                // Create a new session token if one doesn't exist - REUSE SESSION
+                if (!sessionTokenRef.current) {
+                    sessionTokenRef.current = new AutocompleteSessionToken()
+                }
+
+                // Create the request
+                const request = {
+                    ...requestOptions,
+                    input: input,
+                    includedPrimaryTypes: ['street_address'],
+                    sessionToken: sessionTokenRef.current // REUSED SESSION TOKEN
+                }
+
+                // Add included region codes for country-specific results as requested in code review
+                // Places API (New) uses includedRegionCodes instead of componentRestrictions
+                if (countryCode) {
+                    request.includedRegionCodes = [countryCode]
+                }
+
+                // Get suggestions from Google Maps API
+                const response = await AutocompleteSuggestion.fetchAutocompleteSuggestions(request)
+
+                // Convert Google Maps format to our expected format
+                const googleSuggestions = convertGoogleMapsSuggestions(response.suggestions)
+
+                setSuggestions(googleSuggestions)
+            } catch (error) {
+                console.error('Error fetching address suggestions:', error)
```

No console please, let's fail silently